### PR TITLE
[libu2f-server] New Port

### PR DESCRIPTION
**Pull request by xgcssch**
Documentation can be found here at [Yubico](https://developers.yubico.com/libu2f-server/)

### CODE REVIEW

**Comment by xgcssch** [comment]
In the CI pipeline, the platforms x64-windows-static, x64-uwp and arm-uwp fail, despite building fine on a local machine.
It seems, that the includefiles for the dependencies jason-c and openssl are not installed.
I have no clue, why this happens.

**Comment by dan-shaw** [comment]
I've changed it so we don't use msbuild, and it appears that it passes on Linux.

File: ports/libu2f-server/windows.patch [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+diff --git a/u2f-server/u2f-server.h b/u2f-server/u2f-server.h
+index 9a72e54..e367647 100644
+--- a/u2f-server/u2f-server.h
++++ b/u2f-server/u2f-server.h
+@@ -129,6 +129,9 @@ extern "C" {
+ 
+   void u2fs_free_auth_res(u2fs_auth_res_t * result);
+ 
++  u2fs_rc encode_b64u( const char *data, size_t data_len, char *output );
++  u2fs_rc u2fs_get_challenge( u2fs_ctx_t * ctx, const char **challenge );
```
Comment (ras0219-msft) [inline ports/libu2f-server/windows.patch]: This looks like a "feature" -- why do we need these new functions patched in?

If these are feature changes, they should be pushed upstream and go through the normal OSS review process (see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-implement-features-in-patches)

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**